### PR TITLE
MNT: Compat with pytest 8.1

### LINKED ISSUE #216: Upgrade to new-style hook wrappers
Pytest 8 will use the [new-style hook wrappers](https://pluggy.readthedocs.io/en/stable/#wrappers) so we should add a version check to update the decorator for our `pytest_runtest_call` hooks. 

See also: https://github.com/pytest-dev/pytest/issues/11714

Also re-enable the pytestdev test environment.

### PR DESCRIPTION
**Pull request by pllim**
To avoid this in pytest 8.1.dev:

```
pluggy._manager.PluginValidationError: Plugin 'pytest_mpl' for hook 'pytest_report_header'
hookimpl definition: pytest_report_header(config, startdir)
Argument(s) {'startdir'} are declared in the hookimpl but can not be found in the hookspec
```

See https://github.com/pytest-dev/pytest/issues/11779#issuecomment-1879387034

Fix #216 , xref https://github.com/pytest-dev/pytest/issues/11714#issuecomment-1864962140

### CODE REVIEW

**Review by bsipocz — COMMENTED** [review]
Could you also revert https://github.com/matplotlib/pytest-mpl/commit/bd7b5c61863170c809c182c9a3bcd81697b52f1a, so this could be tested?

(and suggestion for the maintainers: add a job for testing with the 8.0.x branch (RC) just as well with pytestdev)

**Comment by pllim** [comment]
Good catch; reverted. Thanks!

**Comment by pllim** [comment]
I have no idea what this test failure is in pytest-dev job

```
E                   ValueError: I/O operation on closed file

FAILED ../../tests/test_baseline_path.py::test_config[None-None-dir3-dir3-True]
```

**Comment by pllim** [comment]
That commit I reverted was from #215 by @ConorMacBride . I have no idea what he meant by "Skip pytestdev until hook wrapper issue is fixed" as there is no further explanation in the PR nor any linked issue.

**Comment by ConorMacBride** [comment]
The hook wrapper issue is #216

File: pytest_mpl/plugin.py [hunk 1]
```diff
@@ -128,11 +132,19 @@ def wrapper(*args, **kwargs):
         item.obj = figure_interceptor(plugin, item.obj)
 
 
-def pytest_report_header(config, startdir):
```
Comment (ConorMacBride) [inline pytest_mpl/plugin.py:131]: Can we just remove all the arguments as none of them are used in the hook?

Comment (pllim) [reply pytest_mpl/plugin.py:131]: I can try that.

Yes, I opened this PR because it was failing in astropy devinfra job.

**Comment by pllim** [comment]
pytestdev is green but the log has traceback, is this expected? I don't maintain this plugin so I don't know what the test is supposed to do.

https://github.com/matplotlib/pytest-mpl/actions/runs/7464235520/job/20310772695?pr=219

**Comment by pllim** [comment]
Would be nice to get this merged sooner than later if this patch is acceptable. astropy devinfra job is failing without this.

**Review by ConorMacBride — APPROVED** [review]
Thanks @pllim and @bsipocz!

**Pull request merged**